Dr Rosaria Lena

Room 620, Kelvin Building

<rosaria.lena@glasgow.ac.uk>

-----------


# Look at the night sky, what do we see?

Let's start with some simple observations of the night sky...
"Look at the night sky, what do we see?"
Here in Scotland, the answer to this questions is likely 'clouds', but if you are curious to know what is beyond those clouds, you can check out Stellarium (you can download it or you can use the web version: https://stellarium-web.org/).

All that glitters is not (only) stars!

![An example from Stellarium](img/Stellarium_full.png)

Looking at the night sky we may be able to see planets, meteors, galaxies, satellites, comets, star clusters and many stars!

![An example from Stellarium (with marked typical objects)](img/Stellarium_full_description.png)

# What is a star?
A star is an astronomical spheroid body:
- held together by _self-gravity_
- _radiating_ energy from an internal source (tipycally from nuclear fusion reactions, and occasionally from the release of gravitational potential energy during contraction or collapse)

Having this definition you should be able to answer this question: "is the Death Star a (fictional) star"?

The answer is "no", because it does not radiate energy! The Death Star is actually a (fictional) space station orbiting the (fictional) planet moon of Endor.
Enough with the fictional stuff now.

If you look at the sky you may see that planets and comets are also shining in the sky, so why can't they be considered stars, based on the definition given above? Do they radiate energy? Are they self-gravitating?

Planets actually do not _radiate_, they mostly reflect the light coming from the Sun, and the same holds for comets. In addition to that, comets do not have enough mass to be considered self-gravitating objects.

## Life and death of a star
A star is not a static object. During its life it will evolve, changing its internal structure and chemical composition as it 'burns its _fuel_' and changing its mass and size, eventually resulting in its death when one of the conditions that define a star is no longer valid, i.e. when the star is no longer self-gravitating or when it stops radiating having exhausted its nuclear fuel. We will talk more about the evolution of stars in the following lectures.
Now let's go back to what we can observe having a look at the night sky and let's see what we can tell from observations of stars.

## Do all stars look the same?
<img src="img/intro2.png" width="600" height="600"/>

![M3](img/M3_color3.gif)

- There are a lot of stars, not uniformly distributed
- Some look brighter than others
- Some are bigger than others
- They have different colours
- Sometimes there are coloured or dark areas around stars
- Some stars have fluctuating brightness: repeating and irregular

## A comparison between two stars on Stellarium

Let's check the properties of two stars on Stellarium. What can you notice after a first look at these stars? You may notice that they have different colours (one looks white, the other one looks white-blue). The boxes contain some information about these stars, and you may recognise many of these fields in the blue boxes from positional astronomy. We will find out what the properties in the red boxes mean during this course.

![Comparison between two stars on Stellarium](img/Stellarium_stars_p.png)

## What can we tell from observations?

- How distant are the stars?
- How big are they?
- How bright?
- How, when and where do they form?
- What are their most important characteristics?
- What are they made of?
- What is their energy source?
- What happens when it runs out?

# How distant are the stars?

## An easy case: the distance to the Sun

__How do we find the distance to the Sun?__

Easy! - Just define The Astronomical Unit (a.u.) as the average distance between the Earth and the Sun!

## The astronomical unit

Ok, but how big is an au?
In 1976 the International Astronomical Union (IAU) adopted a standard definition whose value has been updated with increasing precision  during the years. At the current date, the astronomical unit is:
- 1 a.u. = 149,597,870,700 m

### Question

Can you now work out how long it takes for the sunlight to reach Earth?


In [1]:
import warnings
warnings.filterwarnings('ignore')
import math
AU = 149597870 # km
c = 299792.458 # km/s
ts = AU/c # s
tm = ts/60 # min
print(f"Time taken by light to reach Earth from the Sun = {ts:1.4} s = {tm:1.4} min = 8' 19'' ")

Time taken by light to reach Earth from the Sun = 499.0 s = 8.317 min = 8' 19'' 


### A remark on the Astronomical Unit

Note: 1 a.u. is an _average_ distance, because the distance between the Earth and the Sun varies during the year.

<!-- ![au_remark](img/Earth-orbit-Sun_fixed.png) -->

<div>
<img src="img/Earth-orbit-Sun_fixed.png" width="700" height="500"/>
</div>


In [2]:
import warnings
warnings.filterwarnings('ignore')
import math
AU = 149597870 # km
aphD = 152100000 # km
perD = 147100000 # km
AU_avg = (aphD+perD)/(2*AU) # Average distance, gives AU, check: print(f"{AU_avg}")
print(f"Distance Earth-Sun at aphelion (most distant) = {aphD:1.3e} km = {aphD/AU:1.4} au")
print(f"Distance Earth-Sun at perihelion (closest) = {perD:1.3e} km = {perD/AU:1.3} au")
print(f"Average distance Earth-Sun = {AU:1.3e} km = 1 au")

Distance Earth-Sun at aphelion (most distant) = 1.521e+08 km = 1.017 au
Distance Earth-Sun at perihelion (closest) = 1.471e+08 km = 0.983 au
Average distance Earth-Sun = 1.496e+08 km = 1 au


In [3]:
from IPython.display import display, HTML
display(HTML("<style>.container { width:100% !important; }</style>"))

## Size of our solar system


| Planet | Distance from Sun (a.u.) |
| ------ | ----------------- |
| Mercury | 0.39 |
| Venus | 0.72 |
| Earth | 1 |
| Mars | 1.52 |
| Jupiter | 5.20 |
| Saturn | 9.54 |
| Uranus | 19.2 |
| Neptune | 30.06 |

<img src="img/Solar_system_distances.png" style="max-height: 300px; base64; display: block; position:relative; float:centre; padding-right:0.5cm"/>
    

## How do we measure the distance to the stars?
### The parallax method

We started with the easy example of answering how distant is the Sun, but what about other stars? One way to determine the distance to nearest stars is to use the _parallax method_.

If you look at the position of a closer object against a further background, you will notice that the image of the near object will shift in the background as you move. This effect is called parallax and you can experience it even when you close one eye at the time looking at an object against a background.

<!-- <div>
<img src="img/Parallax_Example.png" width="700"/>
</div> -->
![Parallax1](img/Parallax_Example.png)

Analogously, in a background of distant stars, a closer star will appear to move against them as the Earth orbits the Sun. So how do we use this to determine the distance to a start? We use trigonometry.

The angle formed between the lines of sight of a star from two positions of the observer is twice the parallax angle.
In the example below, the parallax is the angle between the lines of sight of the observer on Earth and from the Sun (the middle point between the position of Earth six months apart).

The number of stars that can be observed from Earth is limited and nowadays _direct_ observations of the distance using the parallax method are done by satellites orbiting the Sun in sync with Earth. Hipparcos gathered data on more than one million stars, and its successor Gaia has already observed 2 billion stars!

<!-- <img src="img/parallax_parsecs.webp" style="max-height: 600px; display: block; position:relative; float:right; padding-left:.5cm"> -->

![Parallax and parsecs representation. Image readapted from: https://www.astronomy.com/wp-content/uploads/sites/2/2023/02/parsec.png](img/parallax_parsecs.png)

<!-- $\begin{equation} -->
    
<!-- \end{equation}$ -->

- The parallax is generally measured in arcseconds (_as_). 1 arcsecond $= \dfrac{1}{3600} \deg = \dfrac{1}{206,265}$ radians. You can show these conversions by recalling that $1\ \rm{rad} = 57.3\deg = (57.3\times 60\times 60)\ \rm{as}=206,265\ \rm{as}$.

- The parsec (**par**allax-**sec**ond, _pc_) is defined as: $$ 1\ \rm{parsec} = \dfrac{1 au}{\rm{arcsecond\  in\  radians}} = 206,265\ au= \dfrac{648000}{\pi} \rm{au} \approx 3.086\times 10^{16}\rm{m}$$.

- A parallax of 1 arcsecond corresponds to a distance to the star of 1 parsec (3.26 light-years).

- Using trigonometry, the distance $d$ to the star is given by:

$$
    d =\dfrac{1 \rm{au}}{\tan p}\simeq \dfrac{1}{p} \rm{au} 
$$
$\qquad$ where $p$ is the parallax, and for this approximation we used $p$ measured in radians.
Converting the parallax $p$ to $p''$, from radians to arcseconds, we obtain the distance in parsec:
$$
    d \simeq\dfrac{206,265 \rm{au}}{p''}= \dfrac{1}{p''} \rm{pc} 
$$


In [4]:
import warnings
warnings.filterwarnings('ignore')
import numpy as np
import math
AU = 149597870 # km
pc = 648000/np.pi*AU # km
print(f"1parsec = {pc:1.3e} km")

1parsec = 3.086e+13 km


### The nearest star: Proxima Centauri

The further a star is, the smaller the parallax angle.

The nearest star (Proxima Centauri) has a parallax of 0.7681 arcseconds. This is _very_ tiny, but still, we are able to measure it! To give you an idea of how small this is:
- It is approximately the angle subtended by an object having a diameter of 2 centimeters, positioned at a distance of 5.3km!
- The shift produced by the parallax against the background would be $1/2000^{th}$ the width of the Full Moon!

The parallax method is covered in the 'Observational Methods' course, so this is just a taster, but now knowing the parallax of Proxima Centauri (or another star), we can find the distance to that.


In [5]:
# The correct version
import warnings
warnings.filterwarnings('ignore')
import math
import numpy as np
def arcsec_to_rad(p):
    rad = p/206265
    return rad

p = 0.7681 # as
d_pc = 1/p # pc

p_rad = arcsec_to_rad(p)
d_au = 1/p_rad # au
d_km = d_au*AU # km
print(f"p = {p} as")
print(f"d = 1/p pc = 1/{p} pc = {d_pc:1.1e} pc")
print(f"d = 1/p au = 20625/p au = 20625/{p} au = {d_au:1.1e} au")
print(f"d = {d_au:1.1e} au * 149597870 km = {d_km:1.1e} km")
print(f"The distance from Proxima Centauri is = {d_pc:1.1e} pc = {d_au:1.1e} au = {d_km:1.1e} km")

p = 0.7681 as
d = 1/p pc = 1/0.7681 pc = 1.3e+00 pc
d = 1/p au = 20625/p au = 20625/0.7681 au = 2.7e+05 au
d = 2.7e+05 au * 149597870 km = 4.0e+13 km
The distance from Proxima Centauri is = 1.3e+00 pc = 2.7e+05 au = 4.0e+13 km


### 'Spot the mistake' question:
An unknown star has a parallax of 379.21 mas. Your messy lecturer tried to calculate the distance to the star but she must have made some mistakes because something looks wrong. Can you tell what?

__Hints__

<details>
<summary>Hint 1</summary>
Check the orders of magnitude, do they seem reasonable?
<details>
<summary>Hint 2</summary>
Compare these orders of magnitude with the ones found for Proxima Centauri, any red flags?
<details>
<summary>Hint 3</summary>
The units are important!
</details>
</details>
</details>


```python
# This is the version with mistakes
import math

p = 379.21/10000 # as 
d_pc = 206265/p # pc

d_au = 1/p # au

AU = 149597870 # km
d_km = d_au*AU # km
print(f"The distance to the star is = {d_pc:1.1e} pc = {d_au:1.1e} au = {d_km:1.1e} km? Something does not look right... find the mistakes!")
```
The distance to the star is = 5.4e+06 pc = 2.6e+01 au = 3.9e+09 km? Something does not look right... find the mistakes!


<!-- ```{python}
# The version with mistakes, where I made a wrong units conversion and I swapped the formulas for d in pc and in au
import warnings
warnings.filterwarnings('ignore')
import math

p = 379.21/10000 # as 
d_pc = 206265/p # pc

d_au = 1/p # au

AU = 149597870 # km
d_km = d_au*AU # km
print(f"The distance to the star is = {d_pc:1.1e} pc = {d_au:1.1e} au = {d_km:1.1e} km? Something does not look right... find the mistakes!") -->
```

<details>
<summary> __Solution__

Below is the solution to the question. Something was clearly wrong with the orders of magnitudes of the values obtained in the wrong solution:
- the distance in pc can not be larger than the distance in au! Those orders of magnitude are clearly off!
- if you compare this with the values obtained for Proxima Centauri, you should expect approximately the same orders of magnitude
- $10^6$ pc would be way too far to be able to use the parallax method to measure the distance: the most precise and reliable measurements of the parallax (with the Hubble Space Telescope and Gaia space mission), can measure distances of the order of $10^3\rm{pc}$.

<!-- ```python 
# The correct version
import warnings
warnings.filterwarnings('ignore')
import math

p = 379.21/1000 # as <- the units conversion was wrong
d_pc = 1/p # pc <- This formula for the distance in pc was swapped with the one below for the distance in au

d_au = 206265/p # au <-

AU = 149597870 # km
d_km = d_au*AU # km
print(f"The distance to the star is = {d_pc:1.1e} pc = {d_au:1.1e} au = {d_km:1.1e} km
```
The distance to the star is = 2.6e+00 pc = 5.4e+05 au = 8.1e+13 km -->


In [6]:
# The correct version
import warnings
warnings.filterwarnings('ignore')
import math

p = 379.21/1000 # as <- the units conversion was wrong
d_pc = 1/p # pc <- This formula for the distance in pc was swapped with the one below for the distance in au

d_au = 206265/p # au

AU = 149597870 # km
d_km = d_au*AU # km
print(f"The distance to the star is = {d_pc:1.1e} pc = {d_au:1.1e} au = {d_km:1.1e} km")

The distance to the star is = 2.6e+00 pc = 5.4e+05 au = 8.1e+13 km


</details>

## Transit method

<img src="img/HalGreg.jpg" style="max-height: 200px; display:block; position:relative; float:left; padding-right:1cm">
The parallax was also originally proposed as a way to determine the distance to the Sun by the Scottish astronomer James Gregory, in 1663, using the transit of Venus. Gregory is seldom given any credits for this idea and usually all the credits goes to Edmund Halley for some reason (can anyone find out why?).

In 1677, at the age of 20, Halley travelled to St Helena to map the Southern Skies and observed a transit of Mercury on November 7th.
Like Gregory, he suggested that the transit of a planet like Venus could be used as a method to measure the distance to the Sun, using parallax.

Here is the translation of Halley's publication about this idea: https://eclipse.gsfc.nasa.gov/transit/HalleyParallax.html

<!-- ![Transit of Venus](img/VenusTransit.png) -->


<img src="img/VenusTransit.png" style="max-height: 500px; display: block; position:relative; float:left; padding-right:1cm">


![Transit Method](img/TransitMethod.png)

According to 'Halley's method', two observers (A and B in the figure above) have to be positioned at a great distance from each other, but both located on the side of Earth facing the Sun at the time of the transit. Each observer would precisely measure the times that Venus appear to be in different positions (contact points) relative to the Sun. The observers would see different chords of the planet's trajectory across the disk of the sun due to parallax. 
Venus forms two similar triangles with the Earth and the Sun at their opposite edges. Since the triangles are similar, the projected distance on the Sun can be found by comparing it to a known baseline on Earth. Once the angle shift is measured, the distances to Venus and then to the Sun can be determined using trigonometry.

Modern techniques use radar echo ranging to measure distances in the Solar System very precisely.

<!-- Venus forms two similar triangles with the Earth and the Sun at their opposite edges. Since the triangles are similar, the projected distance on the Sun can be found by comparing to a known baseline on Earth. -->

# How big are stars?

<img style="float: right; max-height: 350px; position:relative;" src="img/AngularDiameter.png">



## How big is the Sun?

We can deduce the width of the sun if we know its angular diameter $\alpha$ and the distance $D$.

For small angles, $\alpha \approx \frac{2R_\odot}{D}$.

WARNING: This formula uses _radians_ for $\alpha$


Since the Sun's angular diameter is $\alpha\approx 0.5^\circ \approx 32 \ \rm{arcmin} \approx 9.3\times 10^{-3} \rm{rad}$, and we know its distance (from Earth) $D = 1 \rm{au} = 149,597,870 \ \rm{km}$, we can now calculate the size of the Sun.


In [7]:
import warnings
warnings.filterwarnings('ignore')
import math
AU = 149597870 # km
ang_diameter_arcm = 32 # angular diameter of Sun is about 32 minutes of arc
ang_diameter = ang_diameter_arcm/60 * math.pi/180 
# ang_diameter = ang_diameter_rad
diameter = AU * ang_diameter
print(f"1 au = {AU:1.3e} km")
print(f"angular diameter of the sun =  {ang_diameter:.3e} radians")
print(f"Diameter of Sun = {diameter:.0f} km")
print(f"Radius of Sun = {diameter/2:.0f} km")

1 au = 1.496e+08 km
angular diameter of the sun =  9.308e-03 radians
Diameter of Sun = 1392520 km
Radius of Sun = 696260 km


The accepted radius is $6.957\times 10^5$ km.

## What about other stars?

Here is a video showing a comparison between star sizes: https://www.youtube.com/watch?v=HEheh1BH34Q
<iframe width="560" height="315" src="https://www.youtube-nocookie.com/embed/HEheh1BH34Q?si=BQb-tLiTvei87S7Q" title="YouTube video player" frameborder="0" allow="accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture; web-share" allowfullscreen></iframe>

### A comparison of some star sizes

Here is a list of some selected examples of star radii, compared to the Sun.
Some stars, such as Sirius B, are much smaller than the Sun. (The picture below shows an artistic impression of Sirius A and Sirius B.)

The typical range of star radii is roughly $10^{-3}R_\odot < R < 10^3 R_\odot$.

| Name | Radius |
|------|--------|
| Sirius B | 0.008 $R_\odot$|
| Sirius A | 1.71 $R_\odot$|
| Arcturus | 12 $R_\odot$ |
| Aldebaran | 22 $R_\odot$ |
| Rigel | 78.9 $R_\odot$|
| Mira | 210 $R_\odot$ |
| Betelgeuse | 320 $R_\odot$ |

<!-- <img src="img/siriusAB.jpeg" style="max-height: 500px; display: block; position:relative; float:right; padding-right:1cm"> -->

![Artistic impression of Sirius A (large blue-white star on the left) and Sirius B (small blue white-dwarf star on the right)](img/siriusAB.jpeg)

_Image credit: NASA, ESA and G. Bacon (STScI)_

- Sirius B is a very small but very hot white dwarf. It is even smaller than Earth, and it cannot be observed by the naked eye. Its companion star Sirius A, in contrast, is the brightest star in the sky
- Betelgeuse – at a distance of 650 l.y. is a red supergiant and rather short-lived star, the second brightest in Orion and the ninth brightest in the sky
- It’s close to the end of its life and en-route to a rather spectacular death within  a million years or so
- For Betelgeuse, the angular diameter $\alpha$ is about 0.000014 degrees ($1.4\times 10^{-5}$ degrees)
- Most stars have a much smaller angular diameter and cannot be directly imaged
- We have to deduce their size indirectly! (more on this later)

![Betelgeuse](img/Betelgeuse.png)

_Image credit: ALMA (ESO/NAOJ/NRAO)/E. O'Gorman/P. Kervella_

## A handy guide to estimating angular diameters

You can estimate angular diameters using your hand! Extend your arm, place the hand at a right angle and follow the intstructions of the following figure to estimate different angles.
This may not be useful to determine the size of objects that are very far and appear too small, such as stars, but can be useful to measure the angular distance between two stars.

![How to use hands to measure angular distances in the sky](img/measuring-sky-with-hand.png)

## How far apart are the stars?

Let's have a look at some orders of magnitude (this is a good resource if you want to visualise them: https://scaleofuniverse.com/en-gb).

- The nearest star (Proxima Centauri) is about $10^{16}$ m away.
- Diameter of the Sun ~ $10^{9}$ m.
- The difference in scale is $10^7$m in distance, or $10^{21}$m$^3$ in volume.
- Relative to their size, stars are the most isolated structures in the universe.
- If stars were grains of sand (~1 mm), they would be ~10 km apart!
- However, we find them in clusters and binaries - what explains this? (more on this later)

# How bright are stars?

If we try to answer this question based on our observations, we have to keep in mind that the apparent brightness of a light source will vary depending on its distance from the observer. The brightenss we observe is therefore not an _intrinsic_ property of the stars, as it depends on the distance between the observer and the stars.

This is the reason why we need to differentiate between the _apparent brightness_ of a star and its _luminosity_.

## Apparent brightness and luminosity

The __apparent brightness__, $F$  (units [$\rm{W/m}^2$]), also referred to as the _radiant flux_ or _power flux_, is the total amount of energy emitted by a star, per unit time, crossing an unit area. It is _not_ an intrinsic property of a star, because it depends on the distance from the star.
 
The following figure can give you a better understanding of that. Each square represents an unit area. The lightbulb, like a star, radiates equally in every direction (i.e. isotropically). The 'rays' coming out of the light source represent the flux. The farther the position of the observer is from the source, the less flux the observer, in one unit area, will receive. The relationship between the power flux, $F$, and the distance to the source, $r$, is given by the inverse square law: $$F\propto \dfrac{1}{r^2}.$$
Since the light source radiates isotropically (even though the figure below shows only a portion of the rays), the power flux $F$ is equal across all parts of the sphere.

![Inverse square law of the radiant flux of a light source](img/inverse_square_law.png)

The __Luminosity__ of a star, $L$, measured in Watts (W) or equivalently Joules per second (J/s), is its energy output per unit time. The luminosity is the _total_ energy radiated in all the directions in an unit time and it does not depend on the distance from the observer: it is an intrinsic property of the star.

How are these two quantities related?

Imagine a sphere of radius $r$ centered on a star, which is isotropically radiating. Since the power flux $F$ is equal across all parts of the sphere and is the energy output, per unit time, crossing an unit area, we need to divide the luminosity equally over the area of the sphere, $4\pi r^2$. Therefore
\begin{equation}
F = \frac{L}{4\pi r^2}
\end{equation}

The apparent brightness depends on the luminosity and on the distance of the observer from the star. From the formula above, you can see how two stars of equal luminosity, located at different distances from Earth, would have different apparent brightness.

_Photometry_ is the measurement of the quantity of light from a star, i.e. the flux of energy per unit time, per unit area, that we receive at Earth.

For a particular star, if we know the luminosity ($L$) of the star and its distance from Earth ($r$) we can calculate the flux using the equation above.

### The Sun's luminosity

For our Sun, we have:

$F_\odot = 1361\,  \mathrm{W/m^2}$ (The "Solar Constant", or "Solar irradiance", sometimes indicated with the symbol $S_\odot$ in textbooks)

$r = 1\, \mathrm{a.u.} \approx 1.5\times 10^{11}\, \mathrm{m}$

so

$L_\odot = 4\pi r^2 F = 3.8275 \times 10^{26}\, \mathrm{W}$


In [8]:
import scipy
from scipy.constants import pi, au, parsec

F = 1361.0 # W/m/m
r = au # m
L = F*4*pi*r**2

print(f'Power flux: F = {F} W/m/m')
print(f'r = 1au = {r:1.4e} m')
print(f'Luminosity: L =  F*4*pi*r**2 = {L:1.4e} W')

Power flux: F = 1361.0 W/m/m
r = 1au = 1.4960e+11 m
Luminosity: L =  F*4*pi*r**2 = 3.8275e+26 W


For other stars there is a large range of luminosities:

$10^{-4}\,L_\odot < L < 10^{6}\,L_\odot$

If we think of the least luminous star as a 100W incandescent lightbulb, this huge range tells us that the most luminous star would be like the entire world’s power output

Future question: __Why is there this very large range?__

### 'Spot the mistake' question
__What would the apparent brightness be for a star like the Sun at a distance of 10 pc?__
Something does not seem to make sense in the answer below, why?


In [9]:
# This is the version containing a mistake
L = 3.8275E26 # W
r = 10 #pc
F = L/(4*pi*r**2)
print(f'Solar luminosity = {L} W')
print(f'Distance r = {r} pc')
print(f'Apparent brightness of the Sun at 10 parsec: F = {F:1.4e} W/m/m')

Solar luminosity = 3.8275e+26 W
Distance r = 10 pc
Apparent brightness of the Sun at 10 parsec: F = 3.0458e+23 W/m/m


<!-- ```python
# This is the version containing a mistake
L = 3.8275E26 # W
r = 10 #pc
F = L/(4*pi*r**2)
print(f'Solar luminosity = {L} W')
print(f'Distance r = {r} pc')
print(f'Apparent brightness of the Sun at 10 parsec: F = {F} W/m/m') -->
<!-- ```
Solar luminosity = 3.8275e+26 W
Distance r = 10 pc
Apparent brightness of the Sun at 10 parsec: F = 3.0458e+23 W/m/m
``` -->
Does this make sense?

__Hints__

Hint 1: compare this apparent brightness obtained below with the value of the power flux used (the Solar constant) in the previous example. What is wrong?

Hint 2: Do a 'sanity check' of the units in the quantities used above. Does it seem correct?

If we compare this value obtained for the apparent brightness, $F=3.046\times 10^{23} \mathrm{W/m}^2$, with the Solar constant used in the previous example, $F_\odot = 1361 \mathrm{W/m}^2$, we can immediately notice that something is wrong. If we are observing the Sun from a further position of 10pc instead of 1au, we cannot expect it to appear brighter!

Let's do a dimensional analysis 'sanity check' by writing down the units of the quantities given above, and by replacing them in the formula to determine the flux from the luminosity:
$$ F=\frac{L}{4\pi r^2} \implies [W/m^2] = \frac{[W]}{[pc]^2} $$
This does not return an identity since $[m]\neq [pc]$, so now we know where made a mistake, by just checking the units! To calculate the value of the flux correctly, you need to convert the distance from parsecs to meters, because the flux should be written in units of [$W/m^2$]!

``` python
# Correct version
import scipy
from scipy.constants import pi, parsec

L = 3.8275E26 # W
r = 10*parsec #pc
F = L/(4*pi*r**2)
print(f'Solar luminosity = {L} W')
print(f'Distance r = 10 pc = {r:1.4e} m')
print(f'Apparent brightness of the Sun at 10 parsec: F = {F:1.4e} W/m/m')

Solar luminosity = 3.8275e+26 W
Distance r = 10 pc = 3.0857e+17 m
Apparent brightness of the Sun at 10 parsec: F = 3.1989e-10 W/m/m
```


In [10]:
# Correct version
import scipy
from scipy.constants import pi, parsec

L = 3.8275E26 # W
r = 10*parsec #pc
F = L/(4*pi*r**2)
print(f'Solar luminosity = {L} W')
print(f'Distance r = 10 pc = {r:1.4e} m')
print(f'Apparent brightness of the Sun at 10 parsec: F = {F:1.4e} W/m/m')

Solar luminosity = 3.8275e+26 W
Distance r = 10 pc = 3.0857e+17 m
Apparent brightness of the Sun at 10 parsec: F = 3.1989e-10 W/m/m


__Sanity checks__

Get the habit of doing 'sanity checks' of the dimensional analysis and units when you use or derive formulas. This will help you spot some mistakes quickly and is incredibly useful to rederive some formulas sometime. In the formula above for instance, we know that the apparent brightness $F$ has units of [W/m $^2$] and that the luminosity $L$ has units of [W]. Writing the units of each side of Equation (1) we see that:
$ [W/m^2]=\dfrac{[W]}{[m^2]}$, which as expected satisfies the identity, meaning that the formula is dimensionally correct.

## Stellar magnitude scales

The flux from even nearby stars is tiny, but many bright stars are distinguished by the naked eye. The human eye (like hearing) responds logarithmically to stimulus. When comparing the flux densities of three stars in a 1:10:100 ratio, the perceived brightness gap between the first and second star appears to be the same as the difference between the second and third star. This equality in perceived brightness gaps illustrates that human perception of brightness operates on a logarithmic scale. Therefore, before modern photometry, astronomers had a scale based on their own naked-eye observations: __stellar (apparent) magnitude__.

### Apparent magnitude

This system to measure the stars brightness was first introduced by the Greek astronomer Hipparchos, who compiled a catalog of 850 stars with their positions and their _apparent magnitudes_ on a scale from $m=1$, being the brightest stars, to $m=6$, being the faintest stars.
The apparent magnitude depends on the position of the observer the same way as the flux does (in fact, we will see in the following how these two quantities are related).

The image below shows an example of a star catalog created by ancient Greek astronomers.

![Almagest](img/Almagest.png)

_Almagest_ Ptolemy, c. 150AD image credit: Bibliothèque Nationale, Paris

The English translation reads

![Almagest translated](img/AlmagestTranslation.png)

We can still recognise the constellations and find these stars today!

<img src="img/littlebear.svg" style="float: right">

The modern names and magnitudes for these stars are

| Modern Name | Magnitude |
|-------------|-----------|
| Polaris | 1.98 |
|Yildun | 4.35 |
| $\epsilon$ Umi | 4.21 |
| Alasco | 4.95 |
| Ahfa al Farkadain | 4.29 |
| Pherkad | 3.04 |
| Kochab | 2.07 |

<small>Image credit: By IAU and Sky &amp; Telescope magazine (Roger Sinnott &amp; Rick Fienberg) - <a rel="nofollow" class="external autonumber" href="http://www.iau.org/public/constellations/">[1]</a>, <a href="https://creativecommons.org/licenses/by/3.0" title="Creative Commons Attribution 3.0">CC BY 3.0</a>, <a href="https://commons.wikimedia.org/w/index.php?curid=15412502">Link</a>
</small>

Note that some of the magnitudes of the little bear stars have changed since ancient times. Partly this is because the scientific definition and measurements are more precise, but also Polaris is a variable star (its brightness changes).

The apparent magnitude scale introduced by Hipparchos has been replaced by Pogson in 1856 and is no longer limited to values between 1 and 6, it now extends between m=-26.83 for the Sun to over m=30 for the faintest object, but the new classification kept these characteristics:

- 5 magnitude steps correspond to a factor of 100 in apparent brightness
- A difference of 1 magnitude corresponds to an apparent brightness ratio of $100^{1/5}\simeq 2.512$, i.e. $\dfrac{F_{m+1}}{F_m}=100^{1/5}$
- Lower magnitudes are _brighter_
- The zero point of the apparent magnitude has been _conventionally_ determined

<!-- This is not part of the S.I. system you are familiar with, but it is still widespread in astronomy today. We will learn how to use it in our calculations and relate it to S.I. units. -->
The _apparent magnitude_ $m$ is a measure of a star's apparent brightness, i.e. the flux received from the star, and it therefore depends on how close the star is to the observer.

The definition of the apparent magnitude is based on the difference between magnitudes and corresponds to the ratio between two stars' fluxes, as stated in the properties above.

If we consider two stars with measured fluxes (units $W/m^2$) $F_1$ and $F_2$, their apparent magnitudes will be related by

\begin{align}
m_1 - m_2 = -2.5 \log_{10} \frac{F_1}{F_2}
\end{align}

#### Apparent magnitude zero point convention

The values of the apparent magnitudes are determined in relation to a zero point value $m=0$. By convention, this corresponds to an observed flux of $F_0 = 2.518 × 10^{−8} \ \mathrm{W m}^{−2}$. (Reference: IAU 2015 resolution)

The apparent magnitude of a star is therefore conventionally defined as

\begin{align}
m = -2.5 \log_{10}{\frac{F}{F_0}} = -2.5 \log_{10}{F} - 18.997
\end{align}

The apparent magnitude depends on the flux, and therefore on the intrinsic luminosity of the star and its distance ($r$) from Earth.

Using this zero point convention, from the _nominal solar flux_ $F_\odot = 1361 \mathrm{W/m}^2$, we can find that the apparent magnitude of the Sun is $m_\odot=-26.382\ \mathrm{mag}$.

####  Question

Note that the coefficient in the formula for the apparent magnitude is 2.5 and _not_ 2.518! Why 2.5?

__Hint:__

Use the properties of logarithms.

__Solution__:

Based on the characteristics of the apparent magnitude listed above, for two stars with apparent magnitudes $m$ and $m+1$, the ratio between the fluxes is $\dfrac{F_{m+1}}{F_m}=100^{1/5}$.
How can we rewrite this in a form that explicitely shows the difference between the magnitudes, $m-(m+1)$?

\begin{align}
\dfrac{F_{m+1}}{F_m}=100^{1/5}=10^{2/5} &\implies \log_{10}{\dfrac{F_{m+1}}{F_m}}=\log_{10}{10^{2/5}}=\frac{2}{5}\log_{10}{10}=\frac{2}{5} \implies \frac{5}{2}\log_{10}{\dfrac{F_{m+1}}{F_m}} = 1 = (m+1) - m \\
&\implies m-(m+1)=-2.5\log_{10}{\dfrac{F_m}{F_{m+1}}}
\end{align}

You can see how from this derivation the factor 2.5 comes from the ratio 5/2 in the derivation of the formula for the apparent magnitude (see the first passage).

### Magnitude and units

Note that the magnitude is _dimensionless_, so it does not have units (check this quickly doing dimensional analysis: on the right hand side you have the logarithm of a number, no units). However, to remind us that a value (e.g. $m=4.74$) is a magnitude, you can use the notation 'mag' after the numerical value (e.g. $m=4.74$ mag).

**Beware confusing mass and magnitude!**

<!-- Note that the coefficient in the formula above for the apparent magnitude is 2.5 and _not_ 2.518! Why 2.5? Consider that the flux goes as $r^{-2}$, so $r\propto F^{-1/2}$.
Plug into the apparent magnitude formula and the $1/2$ divides the $5$ into $2.5$. -->

<!-- ### Question

Show that if we have two stars with measured fluxes (units $W/m^2$) $F_1$ and $F_2$, their magnitudes will be related by

\begin{align}
m_1 - m_2 = -2.5 \log_10_10_{10} \frac{F_1}{F_2}
\end{align} -->

<!-- __Hint:__

Use the properties of logarithms.

__Solution:__

Having defined the apparent magnitude using the zero point flux above, for the i-th star (i.e. for i=1,2) with flux $F_i$, the apparent magnitude is:
$m_i = -2.5 \log{\dfrac{F_i}{F_0}}$

Therefore:
\begin{align*}
m_1 - m_2 & = -2.5 \log{\frac{F_1}{F_0}} - \left(-2.5 \log{\frac{F_2}{F_0}}\right) \\
& = -2.5\log F_1 - ( -2.5\log F_0) - ( -2.5\log F_2 - ( -2.5\log F_0)) \\
& = -2.5 \log F_1 + 2.5 \log F_2\\
& = -2.5 \log{\frac{F_1}{F_2}}
\end{align*} -->

### Absolute Magnitude

Absolute magnitude (M) is a measure solely of the star's intrinsic luminosity $L$, so it does _not_ depend on its distance to the observer.

It is conventionally defined as the apparent magnitude of a star at a distance $r=10\mathrm{pc}$.

To define the zero point of the absolute magnitude ($M=0$), we use the same value of the flux, $F_0$, used to define the apparent magnitude $m=0$, and we find the corresponding luminosity $L_0$ using Equation (1) for $r=10\mathrm{pc}$.


In [11]:
from scipy.constants import pi, parsec
f0 = 2.518021002E-8 # W/m/m
r = 10*parsec # m
L0 = 4*pi*r**2*f0 # W
print(f'r = 10 pc = {r:1.4e} m')
print(f'Conventional flux for zero apparent magnitude (m=0): F0={f0:1.4e} W/m/m')
print(f'Conventional luminosity for zero absolute magnitude (M=0): L0={L0:1.4e} W')

r = 10 pc = 3.0857e+17 m
Conventional flux for zero apparent magnitude (m=0): F0=2.5180e-08 W/m/m
Conventional luminosity for zero absolute magnitude (M=0): L0=3.0128e+28 W


#### Absolute magnitude zero point convention

The zero value of the absolute magnitude, $M=0$, corresponds to a luminosity of $$ L_0 = 4\pi r^2 F_0 = 3.0128 \times 10^{28} \mathrm{W}.$$ (Reference: <a href="https://www.iau.org/static/resolutions/IAU2015_English.pdf">IAU 2015 resolution </a>)

Plugging Equation (1) into Equation (3), and considering that by definition the absolute magnitude is equal to the apparent magnitude at a distance $r=10\mathrm{pc}$, we can define the absolute magnitude as
\begin{align}
M = -2.5\log_{10}{\frac{L_1}{4\pi r^2}\frac{4 \pi r^2}{L_0}} \implies M = -2.5 \log_{10}{\frac{L_1}{L_0}} = -2.5\log_{10} L_1 + 71.1974
\end{align}

Using this convention, the absolute magnitude of the sun, having a _nominal solar luminosity_ $L_\odot = 3.828 \times 10^{26} \mathrm{W}$, is $M_\odot = 4.74\mathrm{mag}.$


(!) Caution: in different textbooks or sources (not updated to the IAU 2015 resolution), you may find different conventions for the zero points of apparent and absolute magnitudes. Consequently, you may have different values of the apparent and absolute magnitudes of the Sun (or other stars).


In [12]:
from numpy import log10
from astropy.constants import L_sun, L_bol0

print(L_sun)
print(L_bol0)
L0 = 3.0128E28 # W
M = -2.5*log10(L_sun/L_bol0)
print(f'Absolute magnitude of the Sun: M_sun = {M:.2f}')

  Name   = Nominal solar luminosity
  Value  = 3.828e+26
  Uncertainty  = 0.0
  Unit  = W
  Reference = IAU 2015 Resolution B 3
  Name   = Luminosity for absolute bolometric magnitude 0
  Value  = 3.0128e+28
  Uncertainty  = 0.0
  Unit  = W
  Reference = IAU 2015 Resolution B 2
Absolute magnitude of the Sun: M_sun = 4.74


### Setting the Absolute Magnitude scale

We will need to convert magnitudes to S.I. units of luminosity.
In some textbooks you may find that to define the absolute magnitude, the formula above is used, considering the Sun's luminosity instead of the zero value $L_0$ introduced above.

Since the <a link="https://www.iau.org/static/resolutions/IAU2015_English.pdf"> IAU 2015 resolution</a>, this definition is outdated, and the luminosity of a star, in Watts, can be defined using the zero point luminosity previously introduced, $L_0=3.0128\times 10^{28}$ W, as follows.

For a star with luminosity $L$, the absolute bolometric magnitude $M$ is given by
\begin{equation}
L = L_0 10^{-0.4 M}
\end{equation}

#### Example

We can use this formula to recover the luminosity of the Sun previously used, having the absolute magnitude of the Sun, $M_\odot = 4.74$. The constant 0.4 in the formula above is conveniently used to return the Sun's luminosity $L_\odot = 3.826 \times 10^{26}\rm{W}$.


In [13]:
from astropy.constants import L_bol0

M_sun = 4.74
print(L_bol0)
L = L_bol0*10**(-0.4*M_sun)
print(f'Luminosity of the Sun: L = {L:1.3e}')

  Name   = Luminosity for absolute bolometric magnitude 0
  Value  = 3.0128e+28
  Uncertainty  = 0.0
  Unit  = W
  Reference = IAU 2015 Resolution B 2
Luminosity of the Sun: L = 3.828e+26 W


### Question

Show that if we have two stars with luminosities (units $W$) $L_1$ and $L_2$ at the same distance $r=10\rm{pc}$, their difference in the apparent magnitudes is the same as the difference of the absolute magnitudes:

\begin{align}
M_1 - M_2 = -2.5 \log_{10} \frac{L_1}{L_2}
\end{align}

__Hint:__

Use the relationship between flux and luminosity and the properties of logarithms.

__Solution:__

For two stars _at the same distance_ $r$ (e.g. two stars in a binary), we can write
\begin{align}
m_1 - m_2 &= -2.5\log_{10}\left(\frac{L_1}{4\pi r^2} \times \frac{4\pi r^2}{L_2} \right)\\
&= -2.5\log_{10}\frac{L_1}{L_2}
\end{align}

Having defined the absolute magnitude using the zero point flux above, for the i-th star (i.e. for i=1,2) with luminosities $L_i$ at distances of 10pc, the absolute magnitude is:
$M_i = -2.5 \log_{10}{\dfrac{L_i}{L_0}}$

Therefore:
\begin{align*}
M_1 - M_2 & = -2.5 \log_{10}{\frac{L_1}{L_0}} - \left(-2.5 \log_{10}{\frac{L_2}{L_0}}\right) \\
& = -2.5\log_{10} L_1 - ( -2.5\log_{10} L_0) - ( -2.5\log_{10} L_2 - ( -2.5\log_{10} L_0)) \\
& = -2.5 \log_{10} L_1 + 2.5 \log_{10} L_2\\
& = -2.5 \log_{10}{\frac{L_1}{L_2}}
\end{align*}

- For two stars at the same distance of 10pc from the observer, the difference in the apparent magnitude is equal to the difference in the absolute magnitude.

### Relationship between apparent magnitude and absolute magnitude

By using the conventional definitions of the apparent magnitude and absolute magnitude, we can find the relationship between them:

\begin{align}
m &= -2.5\log_{10}{\frac{F}{F_0}} \\
&= -2.5\log_{10}\left(\frac{L}{4\pi r^2}\times \frac{4\pi (10\rm{pc})^2}{L_0} \right) \\
& =-2.5\log_{10}\left(\frac{L}{L_0}\right) -2.5\log_{10} \left( \frac{10\rm{pc}}{r}\right)^2\\
& = M -5\log_{10}(10\rm{pc})+5\log_{10}{r}\\
& = M -5 + 5\log_{10}{r} \implies \\
\implies M &= m + 5 - 5\log_{10} r
\end{align}

(!) Careful to the units! Here $r$ is measured in _parsecs_. Use this formula with caution!


<!-- ## 5.3. <a id='toc5_3_'></a>[Apparent magnitude and luminosity](#toc0_)

If the two objects are at the same distance $r$ (e.g. two stars in a binary) then we can write

\begin{align}
m_1 - m_2 &= -2.5\log_{10}\left(\frac{L_1}{4\pi r^2} \times \frac{4\pi r^2}{L_2} \right)\\
&= -2.5\log_{10}\frac{L_1}{L_2}
\end{align} -->

<!-- ## 5.4. <a id='toc5_4_'></a>[Absolute magnitude and luminosity](#toc0_)
Remember that the absolute magnitude $M$ is equal to the apparent magnitude $m$ at 10 pc distance.

Mathematically, using the relationship between apparent and absolute magnitudes: $M = m +5 -5\log_{10}r$, it follows that

\begin{align}
M_1 - M_2 = -2.5 \log_{10}\frac{L_1}{L_2}
\end{align} -->

### Magnitudes, luminosity and distance

- If we know a star's apparent magnitude $m$ and know its absolute magnitude $M$ you can calculate its distance $r$ (in parsec!) using $M=m + 5 - 5\log_{10} r$.
- If we know a star's apparent magnitude and distance you can compute its luminosity
- You should practice this type of conversion!

### Example

A star with apparent magnitude $4.2$ is measured by parallax to have a distance of $20$ parsecs. What is the absolute magnitude?


In [14]:
from numpy import log10
m=4.2
r=20 #pc
M = m + 5 - 5*log10(r)
print(f'M = m + 5 - 5*log10(r) = {m} + 5 - log10({r})={m}+5{-log10(r)} = {M}')
print(f'Absolute magnitude is {M:.2f}')

M = m + 5 - 5*log10(r) = 4.2 + 5 - log10(20)=4.2+5-1.3010299956639813 = 2.694850021680093
Absolute magnitude is 2.69


### Note on types of magnitude

The magnitude observed can depend on the sensitivity of the telescope or detector to the range of wavelengths. For instance, the human eye can observe in the visible part of the spectrum, but not in the IR or UV. Likewise, telescopes can have filters to observe different parts of the spectrum.
Magnitude can therefore be specified for different wavelength intervals, i.e. observed with different coloured filters on the telescope.
- For stars, the most commonly quoted magnitude is the **visual magnitude**, the magnitude in the visual part of the spectrum
- Another commonly used magnitude is the **bolometric magnitude**, which includes the luminosity across the whole electromagnetic spectrum of wavelengths, whether observed or not. We will use this when performing some physics calculations.

The magnitudes we mentioned so far, e.g. $M_\odot = 4.74$ are __bolometric magnitudes__.

# Recap

- The distance to near stars can be determined using the parallax method
- 1 parsec is the distance to a star having a parallax angle of 1 arcsec

- The apparent brightness of a star scales with the inverse square of the distance to the star
- The flux of a star is the total energy emitted per unit time 

- The magnitude scale is logarithmic
- Apparent magnitude is the magnitude seen from Earth
- Absolute magnitude is intrinsic to the star
- We can find the luminosity in S.I. units from the absolute magnitude
- We can convert apparent to absolute magnitude if we know the distance to the star in parsecs

# Stellar Colour

### Why are the stars different colours?

![starry night](img/intro2.png)

We know that light is an electromagnetic wave, and that the colour we perceive depends on the wavelength of the light.

<img src="https://upload.wikimedia.org/wikipedia/commons/d/d9/Linear_visible_spectrum.svg">

- Red light has a wavelength $\approx 650$ nm
- Blue light has a wavelength $\approx 450$ nm
- White light is a mixture of all visible wavelengths

Stars are tinted with different colours, indicating that their light contains wavelengths more toward the red or blue end of the spectrum.

## Black body radiation

- A body that totally absorbs all radiation that falls upon it is called a __black-body__. You can think of it as a perfect absorber of radiation.
- If it absorbs energy then it will heat up, so a black body can have a temperature $T$.
- If it is hotter than its environment then it will cool down by re-radiating its energy.
- A perfect absorber also emits

What is the distribution of radiation across wavelengths (or frequencies) that a black body emits?

For a perfect emitter, radiation is emitted equally from the surface of an object, and isotropically over solid angle.
We want to know the power emitted per unit wavelength $\lambda$, per unit area, per steradian: the __spectral radiance__ $B_\lambda(\lambda;T)$.
The black body spectrum across frequencies can only depend on the temperature of the object.

Attempts to describe the radiation spectrum of a black body ran into trouble...

### Rayleigh-Jeans law

   
Although the observed spectrum of black-body radiation was known observationally, it proved difficult to explain! This was a puzzle around the end of the 19th Century.
Lord Rayleigh used classical physics to predict the shape of the black body spectrum, this was later refined by Sir James Jeans.

The Rayleigh-Jeans law is a good approximation to the observed spectrum at long wavelength (low frequencies) $hc\ll k_BT\lambda$.

\begin{align}
B_\lambda^\mathrm{RJ}(\lambda; T) = \frac{2ck_B T}{\lambda^4}
\end{align}

<div style="max-width: 20%; float: left; margin: 5mm">
<img src="https://upload.wikimedia.org/wikipedia/commons/2/28/John_William_Strutt.jpg" >
    Lord Rayleigh (1842 - 1918)
</div>
<div style="max-width: 20%; float: right; margin: 5mm">
<img src="https://upload.wikimedia.org/wikipedia/commons/9/92/James_Hopwood_Jeans.jpg" >
    Sir James Jeans (1877 - 1946)
</div>


In [15]:
import numpy as np
from astropy.constants import k_B, c, h
from matplotlib import pyplot as plt
from matplotlib import rc
rc('text',usetex=True)
rc('font',size=15)
rc('figure',figsize=(8,6), dpi=300)

l = np.linspace(0,3000,1000) # Wavelength, nm
T = 4000 # Temperature, Kelvin

B_rj=c.value* 2*k_B.value * T /(l*1e-9)**4 # 1e-9 to convert from nm to meters

In [16]:
plt.plot(l, B_rj, label="Rayleigh-Jeans law")
plt.xlabel(r'Wavelength $\lambda$ (nm)')
plt.ylabel(r'Spectral Radiance $B_\lambda(\lambda, T=4000K)$ (Wm$^{-3}$sr$^{-1}$)')
plt.title('Spectral radiance')
plt.ylim([0,1e15])
plt.grid()
plt.legend();

<Figure size 2400x1800 with 1 Axes>

The Rayleigh-Jeans law has a big problem - the radiance keeps increasing indefinitely at short wavelengths (higher frequencies). If it were true the power emitted at short wavelengths would be infinite! This is known as the Ultraviolet Catastrophe, and indicates the failure of classical physics to explain the behaviour of thermal radiation.

### Wien's law

Wien's law is a good approximation to the observed spectrum at short wavelength (high frequencies) $hc\gg k_BT\lambda$.


\begin{align}
B_\lambda^\mathrm{Wien}(\lambda; T) = \frac{2 hc^2}{\lambda^5}\exp\left(-\frac{hc}{\lambda k_B T}\right)
\end{align}


In [17]:
B_wien=2 * h.value *c.value**2 * np.exp(-h.value*c.value/((l*1e-9)*k_B.value*T)) / (l*1e-9)**5;

In [18]:
plt.plot(l, B_rj, label="Rayleigh-Jeans law")
plt.plot(l, B_wien, label="Wein's law")

plt.xlabel(r'Wavelength $\lambda$ (nm)')
plt.ylabel(r'Spectral Radiance $B_\lambda(\lambda, T=4000K)$ (Wm$^{-3}$sr$^{-1}$)')
plt.title('Spectral radiance')
plt.ylim([0,5e12])
plt.grid()
plt.legend();

<Figure size 2400x1800 with 1 Axes>

## Planck's Law
In 1900, Planck was able to derive a spectrum that fit the data by assuming that energy comes in discrete quanta. This marked the birth of quantum physics! The Planck radiation spectrum for a black body's spectral radiance is:

\begin{equation}
B_\lambda(\lambda; T) = \frac{2 hc^2}{\lambda^5}\left(\frac{1}{\exp\left(\frac{hc}{\lambda k_B T}\right) -1} \right)
\end{equation}


In [19]:
B_planck = 2*h.value*c.value**2 * (1.0/(np.exp(h.value*c.value/(k_B.value*T*(l*1e-9)))-1)) / (l*1e-9)**5;
plt.plot(l, B_planck, label="Planck's law")
plt.plot(l, B_wien, label="Wein's law")
plt.plot(l, B_rj, label="Rayleigh-Jeans law")

plt.xlabel(r'Wavelength $\lambda$ (nm)')
plt.ylabel(r'Spectral Radiance $B_\lambda(\lambda, T=4000K)$ (Wm$^{-3}$sr$^{-1}$)')
plt.title('Spectral radiance')
plt.ylim([0,5e12])
plt.grid()
plt.legend();

<Figure size 2400x1800 with 1 Axes>

### Exercise:

1. Show that for low frequencies ($hc \ll \lambda k_B T$) the Rayleigh-Jeans law can be derived from the Planck law.
1. Show that for high frequencies ($hc \gg \lambda k_B T$) Wein's law can be derived from the Planck law.

You can use the approximation $\lim_{x\rightarrow  0} \exp(x) = 1 + x$

# Temperature and colour

Let's look at how the spectrum changes with temperature $T$.


In [20]:
for T in [6000,5500,5000,4500,4000]:
    B_planck = 2*h.value*c.value**2 * (1.0/(np.exp(h.value*c.value/(k_B.value*T*(l*1e-9)))-1)) / (l*1e-9)**5
    plt.plot(l, B_planck, label='T='+str(T)+'K')

plt.xlabel(r'Wavelength $\lambda$ (nm)')
plt.ylabel(r'Spectral Radiance $B_\lambda(\lambda, T=4000K)$ (Wm$^{-3}$sr$^{-1}$)')
plt.title('Spectral radiance at different temperatures')
plt.grid()
plt.legend();

<Figure size 2400x1800 with 1 Axes>

We notice that the Planck spectrum
- has a clear peak at some wavelength, which decreases as temperature increases
- $B_\lambda(\lambda;T)> 0$ for all wavelengths, but tends toward $0$ at the extreme ends of the spectrum
- overall increases with temperature


Black bodies radiate at all wavelengths - they won't look black! The shape of the spectrum and the position of the peak determines the apparent colour of the object.

## Wien's displacement law


There is an inverse relationship between the temperature $T$ in kelvin and the peak emission wavelength $\lambda_{peak}$ in metres, called Wien's displacement law:
\begin{align}
\lambda_{peak} = \frac{0.0029\,\mathrm{m\,K}}{T}
\end{align}

Where $\mathrm{m\,K}$ are metres $\times$ kelvin, the units of the constant.

We can use this to determine the effective temperature of a star from the spectrum of its light, assuming it is a black body.

## Example

The Sun's spectrum has a peak wavelength of around 500 nm, in the green part of the spectrum.

<img src="https://upload.wikimedia.org/wikipedia/commons/d/d9/Linear_visible_spectrum.svg">

What is the effective emission temperature of the Sun?


In [21]:
w = 0.0029 # constant
lmax = 501e-9 # peak wavelength in metres
Tsun = w/lmax
print('The effective temperature of the Sun is {0} K'.format(Tsun))

The effective temperature of the Sun is 5788.423153692614 K


The commonly used value is $T_\odot=5780$ K

Although the peak emission is greenish, the distribution overall produces light that appears yellowish to us, after passing throught the atmosphere.

## (!) Converting between wavelength and frequency

So far we have been using the spectral radiance in terms of wavelength $B_\lambda(\lambda;T)$. It can also be expressed in terms of the frequency of the radiation $B_\nu(\nu;T)$. Since it is a density across wavelengths, we must be careful when converting between them.

\begin{align}
B_\nu(\nu;T)d\nu &= B_\lambda(\lambda;T)d\lambda \\
B_\nu(\nu;T) &= B_\lambda(\lambda(\nu);T) \left|\frac{d\lambda}{d\nu}\right| \\
 &= B_\lambda(c/\nu; T) \left|-\frac{c}{\nu^2}\right| \\
  &= \frac{2h\nu^5}{c^3}\left( \frac{1}{\exp\left(\frac{h\nu}{k_BT}\right)-1} \right) \times \frac{c}{\nu^2} \\
  B_\nu(\nu;T) &= \frac{2h\nu^3}{c^2}\left( \frac{1}{\exp\left(\frac{h\nu}{k_BT}\right)-1}\right)
\end{align}

## Okay, but how big is an a.u.?

In 1976 the International Astronomical Union (IAU) adopted a standard definition whose value has been updated with increasing precision  during the years. At the current date, the astronomical unit is:
- 1 a.u. = 149,597,870,700 m

## Are stars really black bodies?

- The Planck black body spectrum is a theoretical prediction for an ideal case.
- Light from the Sun is obscured by the atmosphere of the Earth, which absorbs some wavelengths more than others. This is called _atmospheric extinction_. We will ignore this effect in this course.
- There are also _spectral lines_, gaps in the solar spectrum. More on this later.

# Stefan-Boltzmann Law

The total (bolometric) luminosity per area ("radiant emittance" $j=\frac{L}{A}$) has units of $\mathrm{W\,m}^{-2}$, and is the spectral radiance integrated over solid angle and wavelength:

The equation for this is known as the Stefan-Boltzmann Law.
\begin{align}
j &= \frac{\pi^2 k_B^4 T^4}{60 \hbar^3 c^2}
\end{align}

This can be written more simply as

\begin{align}
j = \sigma T^4
\end{align}

where $\sigma = \frac{\pi^2 k_B^4 }{60 \hbar^3 c^2} = 5.67\times 10^{-8}\mathrm{W\,m^{-2}K^{-4}}$ is the Stefan-Boltzmann constant.

# Stellar luminosity

We can multiply by the surface area to find the total luminosity of a spherical star of radius $R$:

\begin{align}
L = 4\pi R^2 \sigma T^4
\end{align}

This is the most useful form to use in stellar calculations - __remember it__!

It allows us to relate $R$, $L$, and $T$ for a star.

- If we know the apparent magnitude and the distance $d$, then we can find the absolute magnitude
- If we know the absolute magnitude we can find the luminosity
- If we know the luminosity and the temperature we can find the star's radius!

You need to be able to solve these types of problem - __practice it__!

## Example

$\epsilon$-Eridani is a star at $r=3.212$ parsecs distance, with an apparent bolometric magnitude of 3.54 and a peak emission wavelength of 570 nm.
What else can we tell about it?

- Work out the absolute bolometric magnitude from the apparent bolometric magnitude

\begin{align}
M_\epsilon &= m_\epsilon + 5 - 5\log_{10}r \\
  &= 3.54 + 5 - 5\log_{10}(3.2) \\
  &= 6.0
\end{align}

- Work out the Luminosity from that and the distance
- Remember solar absolute bolometric magnitude is 4.83

\begin{align}
M_{\epsilon} &= -2.5\log_{10}L_\epsilon+2.5\log_{10}L_\odot + 4.83 \\
2.5\log_{10}(L_\epsilon / L_\odot) &= 4.83 - M_\epsilon \\
&= 4.83 - 6.0  \\
&= -1.17 \\
\log_{10}(L_\epsilon / L_\odot) & = -0.468 \\
L_\epsilon / L_\odot &= 10^{-0.468} \\
\hline \\
L_\epsilon &= 0.34\,L_\odot \\
&= 1.3\times 10^{26}\,\mathrm{W}
\end{align}

- Work out the temperature using Wien's displacement law

\begin{align}
\lambda_{peak} T_\epsilon &= 0.0029 \\
T_\epsilon &= \frac{0.0029}{ 570\times 10^{-9}} \\
\hline \\
T_\epsilon &= 5088\,\mathrm{K}
\end{align}

- Work out the radius

\begin{align}
L_\epsilon &= 4\pi R_\epsilon^2 \sigma T^4 \\
R_\epsilon^2 &= \frac{L_\epsilon}{4\pi \sigma T^4} \\
R_\epsilon &= \sqrt{\frac{L_\epsilon}{4\pi \sigma T^4}} \\
&= \sqrt{\frac{1.3\times 10^{26}}{4\pi \times 5.671\times10^{-8}\times 5088^4}}  \\
\hline \\
&= 5.22\times 10^{9}\,\mathrm{m}
\end{align}

# Recap

- __Planck radiation law__ gives the shape of the black-body emission spectrum
- __Wien's displacement law__ relates the peak emission wavelength to the temperature
- __Stefan-Boltzmann law__ can be used to find the total power radiated by a black body
